query-translator/src/translate.rs

```diff
@@ -117,14 +123,23 @@ pub struct CombinedSelectQuery {
 }
 
 fn cc_to_select_query<T: Into<Option<u64>>>(projection: Projection, cc: ConjoiningClauses, limit: T) -> SelectQuery {
-    SelectQuery {
-        projection: projection,
-        from: FromClause::TableList(TableList(cc.from)),
-        constraints: cc.wheres
-                       .into_iter()
-                       .map(|c| c.to_constraint())
-                       .collect(),
-        limit: limit.into(),
+    if cc.is_known_empty {
+        SelectQuery {
```

Comment:
I decided to push this up to the consuming point — that avoids us creating a projection list.